src/main/scala/db/migration/V25__RemoveDomainFromH5PUrl.scala

```diff
@@ -67,26 +70,54 @@ class V25__RemoveDomainFromH5PUrl extends BaseJavaMigration {
       .apply
   }
 
-  def updateH5PDomains(html: String): String = {
-    val oldDomain = "https?:\\/\\/h5p.{0,8}.ndla.no"
-    val newDomain = ""
+  private def stringToJsoupDocument(htmlString: String): Element = {
+    val document = Jsoup.parseBodyFragment(htmlString)
+    document.outputSettings().escapeMode(EscapeMode.xhtml).prettyPrint(false)
+    document.select("body").first()
+  }
+
+  private def jsoupDocumentToString(element: Element): String = {
+    element.select("body").html()
+  }
+
+  def containsH5PLink(url: String): Boolean = {
+    val h5pUrls =
+      Seq("https://h5p.ndla.no", "https://h5p-test.ndla.no", "https://h5p-ff.ndla.no", "https://h5p-staging.ndla.no")
+    val delUrl = h5pUrls.find(u => url.contains(u))
 
-    val updatedHtml = html.replaceAll(oldDomain, newDomain)
-    updateH5PResourceType(updatedHtml)
+    delUrl match {
+      case Some(_) => true;
+      case None    => false
+    }
   }
 
-  def updateH5PResourceType(html: String): String = {
-    val oldResourceType = "external"
-    val newResourceType = "h5p"
+  def relativeUrl(url: String): String = {
+    url.replaceAll("https?://h5p.{0,8}.ndla.no", "")
+  }
 
-    html.replaceAll(oldResourceType, newResourceType)
+  def updateContent(html: String): String = {
+    val doc = stringToJsoupDocument(html)
+    doc
+      .select("embed")
+      .forEach(embed => {
+        val dataResource = embed.attr("data-resource")
+        if (dataResource == "external") {
+          val url = embed.attr("data-url")
+          if (containsH5PLink(url)) {
+            embed.attr("data-path", relativeUrl(url))
```

Comment:
Validation har ikkje støtte for data-path på embeds så det må du legge til først.